Suggestion for Mvux.Overview (#1584)
unoplatform/uno.extensions: doc/Overview/Mvux/Overview.md

# MVUX Overview

> **M**odel, **V**iew, **U**pdate, e**X**tended (**MVUX**) is a variation of the MVU design pattern that encourages unidirectional flow of immutable data, whilst adding support for the data binding capabilities. And that capabilitie makes the MVVM pattern so powerful. In other words, MVUX brings together the best of the MVU and MVVM design patterns.

Reviewer: I've updated the suggestion slightly as I feel the intent of the original paragraph was lost

> In summary MVUX utilizes its powerful code-generation engine to generate a bindable proxy for each Model, which holds the state information of the data, as well as a bindable proxy for entities where needed.

Reviewer 2: ```suggestion
MVUX utilizes its powerful code-generation engine to generate a **bindable proxy** for Model types which encapsulate state information of the data. Additional proxies may be generated for other entities where needed.
```

> In summary MVUX utilizes its powerful code-generation engine to generate a bindable proxy for each Model, which holds the state information of the data, as well as a bindable proxy for entities where needed.

Reviewer 2: Good point. We could go with phrasing like this instead

> In summary MVUX utilizes its powerful code-generation engine to generate a bindable proxy for each Model, which holds the state information of the data, as well as a bindable proxy for entities where needed.

Reviewer: ```suggestion
MVUX uses a source code generator to generate bindable proxies for each Model. Additional proxies may be generated for other entities where needed.
```

The bindable proxies use as a bridge that enables immutable entities to work with the WinUI data-binding engine. 
The states in the Model are monitored for data-binding changes, and in response to any change, **the objects are recreated fresh**, instead of their properties being changed.

> Thus, with less code typed, we can have a more efficient and practical result.

Reviewer 2: > Thus, with less code typed, we can have a more efficient and practical result.

Let's expand on this line further. Is the result more _efficient_ or _practical_ solely from less developer effort? Maybe delineate between these two aspects.

Reviewer: ```suggestion
```

- Feeds include various dimensions such as loading, if there's data or if an error occurred.
- Feeds borrow from Option concept in functional programming where no data is a valid state for the feed.
- MVUX combines the unidirectional flow of data, and immutability of MVU, with the data binding capabilities of MVVM.

## Learning MVUX by Creating your own

To create a new project using MVUX on Uno Platform see [How to set up an MVUX project](https://github.com/unoplatform/uno.extensions/blob/main/doc/Overview/Mvux/Tutorials/HowTo-MvuxProject.md).

Then you can use the example above as a reference to create your own solution.

As summarized below.

### In the Model

- Create you own Models

> - MVUX recommends using plain POCO (Plain Old CLR Object) record types for the models in your app as they're immutable.

Reviewer 3: ```suggestion
- MVUX as a pattern typically employs [Plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object) ('POCO') record types for the models in your app because of their immutability.
```

> - MVUX recommends using plain POCO (Plain Old CLR Object) record types for the models in your app as they're immutable.

Reviewer: ```suggestion
- MVUX recommends using record types for the Models in your app as they're immutable.
```

- The MVUX analyzers generate a bindable proxy for each of the models in your app (those with Model suffix and mark with the partial modifier). 
- For every public feed property (returning IFeed<T> or IListFeed<T>) found in the model, a corresponding property is generated on the bindable proxy.

### In the View

> - Create your views and add the data binds

Reviewer: ```suggestion
- Create your views and add data binding to the XAML elements as required
```

> - Create and customize your FeedView as you can see in [The FeedView control](https://github.com/unoplatform/uno.extensions/blob/main/doc/Overview/Mvux/FeedView.md)

Reviewer 2: @rafael-rosa-knowcode The page linked did not have a identifier, so you can certainly add one and reference the xref here

> - Create and customize your FeedView as you can see in [The FeedView control](https://github.com/unoplatform/uno.extensions/blob/main/doc/Overview/Mvux/FeedView.md)

Reviewer 3: ```suggestion
- Create and customize your FeedView as you can see in [The FeedView control](xref:Overview.Mvux.FeedView).
```

Added another PR to fix that: https://github.com/unoplatform/uno.extensions/pull/1587

- Or you can use [States](https://github.com/unoplatform/uno.extensions/blob/main/doc/Overview/Mvux/States.md)

### Update

- Add your updates in the Model and in the views

### eXtended

- As the MVUX already generated the bindable proxies.

> - You just need to add an Bindable instance of the `YourModel`, the `DataContext` and it is done.  

Reviewer: ```suggestion
- You just need to create an instance of the bindable proxy for your Model and assign it to the `DataContext` and it is done.  
```
